File: 백준/Silver/11497. 통나무 건너뛰기/통나무 건너뛰기.js
const input = require("fs")
  .readFileSync(process.platform === "linux" ? "/dev/stdin" : "./example.txt")
  .toString()
  .trim()
  .split("\n")
  .map((v) => v.trim().split(" ").map(Number));

const [[t], ...inp] = input;

for (let i = 0; i < t; i++) {
  const [[n], l] = inp.splice(0, 2);

  l.sort((a, b) => a - b);

  let res = 0;
  for (let j = 0; j < n - 2; j++) {
    res = Math.max(res, l[j + 2] - l[j]);
  }
  res = Math.max(res, l[n - 1] - l[n - 2]);

  console.log(res);
}

File: 백준/Silver/11497. 통나무 건너뛰기/통나무 건너뛰기.py
import sys
input = sys.stdin.readline

t = int(input())
for _ in range(t):
    n = int(input())
    l = list(map(int, input().split()))
    l.sort()
    
    res = 0
    for i in range(n-2):
        res = max(res, l[i+2] - l[i])
    res = max(res, l[n-1] - l[n-2])
    
    print(res)